rfcs/protocols/asset-exchange/corda-htlc.md

```diff
@@ -6,45 +6,49 @@
-
-![Corda HTLC](../../resources/images/cordapp-htlc.png)
-
-### Generalisability
-The hash lock encumbrance state and associated contract can be offered as reusable pieces as part of Weaver. These pieces are largely agnostic of the specifics of the application. They can further be instrumented to *emit* custom Weaver events making automation of the orchestration of cross-chain HTLCs possible.
+- Since: 21-Oct-2022
+
+## Generic Corda Network
+Below figure depicts the different components of a Corda network. Primarily it consists of the infrastructure used for the creation of the network and the distributed applications (CorDapps) deployed on the network. Any such CorDapp is a combination of contracts, flows and a client component to interact with the network.
+<img src="../../resources/images/generic-corda-network.png" width=80%>
+
+## Weaver enabled Corda Network
+Below figure depicts the different components of a Corda network with Weaver adaptations (highlighted in _green_ color). It requires changes in both infrastructure and CorDapp (contracts, flows, and client) layers. However, not all these adaptations are needed to demonstrate asset exchange protocol. For example, changes to infrastructure layer (addition of _Relays_) and changes to client layer (addition of _Identity Services_ and _Drivers_) are only needed to demonstrate _data sharing_ and _asset transfer_ protocols.
+<img src="../../resources/images/weaver-enabled-corda-network.png" width=80%>
+
+### Details of adaptations
+The onus on the changes required to made lies both on the application developer (as Weaver is agnostic to the client application) and on Weaver as the platform provider.
+
+#### Responsibilities of CorDapp developer
+
+- Asset contract must be augmented with the ability to verify lock, claim, and unlock requests (using Weaver libraries)
+- Asset contract also should provide a function to create a state using _StateAndRef_ with owner passed as argument.
+- Create following flows to support lock, and claim of an asset (using Weaver SDK):
+  - UpdateOwnerFlow: Takes _StatePointer_, and returns _Asset State_ with updated owner, where caller will be the new owner.
+  - RetrieveStateAndRef: Takes (assetType, id) for non-fungible assets or (assetType, quantity) for fungible assets as arguments, and returns the _StateAndRef_ for the corresponding asset.
+- Add client support:
+  - Endpoints: lock, claim, unlock, and query status of, an asset (using Weaver SDK)
+  - Endpoints: receive hash locks and secrets from corresponding app in the other network
+  - Send: hash locks and secrets to corresponding app in the other network
+
+#### Responsibility of Weaver
+- HTLC states maintained in vaults
+- Interoperation CorDapp to manage bulk of lock, claim, and unlock flows
+
+## Weaver HTLC implementation approach for Corda network
```

Comment:
Move this above the "application adaptation" section.

Reply:
Suggested rename of title: "HTLC Asset Exchange Capabilities in Weaver". "Corda network" is implied by the filename, so I don't think we need to add it to the title.